test/filewatch_test.erl

```diff
@@ -0,0 +1,76 @@
+-module(filewatch_test).
+-include_lib("eunit/include/eunit.hrl").
+
+-define(
+   rapid_events_test(Files, NEvents),
+
+   Dir = create_sandbox(?FUNCTION_NAME),
+   Pairs = [{Dir ++ File, File} || File <- Files],
+   {ok, P} = filewatch:start(self(), Pairs),
+   [start_sender(Path, NEvents) || {Path, _} <- Pairs],
+   State = collect_events(Files),
+   filewatch:stop(P),
+   ?assert(State)
+  ).
+
+one_file_few_rapid_events_test() ->
+    ?rapid_events_test(["testfile"], 5).
+
+one_file_many_rapid_events_test() ->
+    ?rapid_events_test(["testfile"], 1000).
+
+many_files_few_rapid_events_test() ->
+    ?rapid_events_test(["test1", "test2", "test3", "test4"], 5).
+
+many_files_many_rapid_events_test() ->
+    ?rapid_events_test(["test1", "test2", "test3", "test4"], 1000).
```

Comment:
You probably want the `long-named files do not crash` test as well.

Also, for "many files", I recommend generating at least 64k test file names.